openhands/linter/linter.py

```diff
@@ -33,3 +34,33 @@ def lint(self, file_path: str) -> list[LintResult]:
             if res:
                 return res
         return []
+
+    def lint_file_diff(
+        self, original_file_path: str, updated_file_path: str
+    ) -> list[LintResult]:
+        """Only return lint errors that are introduced by the diff.
+
+        Args:
+            original_file_path: The original file path.
+            updated_file_path: The updated file path.
+
+        Returns:
+            A list of lint errors that are introduced by the diff.
+        """
+        updated_lint_error: list[LintResult] = self.lint(updated_file_path)
+
+        with open(original_file_path, 'r') as f:
+            original_file_content = f.read()
+        with open(updated_file_path, 'r') as f:
+            updated_file_content = f.read()
+        diff = get_diff(original_file_content, updated_file_content)
+        changes = parse_diff(diff)
+
+        # Select errors that are introduced by the new diff
+        selected_errors = []
+        for change in changes:
+            new_lineno = change.new
+            for error in updated_lint_error:
+                if error.line == new_lineno:
```

Comment:
This approach would not pass [test_lint_file_catch_new_errors_outside_edits](https://github.com/All-Hands-AI/OpenHands/pull/3649/files#diff-f1e8c3680edd5112543092c9adf55f4cbf79716880a05118b473baf63d7900c8R1509)

Reply:
For example, if the edit changes a function definition, it would cause all callers to fail, even though the edit doesn't touch the callers

Reply:
ah good point.. I'll work on this today to see if we can fix it. If not, how about we merge this first to unblock the editing PR (so other ppl can start working on the FE UI) and then I'll put together a follow-up PR to fix this specific issue?

Reply:
Ahh in this case, we still need to do two passes of lint and map the error from old lines to the new lines 👍  I'll try to tweak that

Reply:
Should be fixed in the new commit!